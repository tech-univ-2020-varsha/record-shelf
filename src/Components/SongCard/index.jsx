
import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const SongCard = ({ albumArtUrl, artists, albumName }) => (
  <div className={styles.songCardContainer}>
    <div>
      <img src={albumArtUrl} alt={albumName} className={styles.albumPicture} />
    </div>
    <div>
      {albumName}
    </div>
    <div>
      {artists.toString()}
    </div>
  </div>
);

export default SongCard;

SongCard.propTypes = {
  albumArtUrl: propTypes.string.isRequired,
  artists: propTypes.arrayOf(propTypes.string).isRequired,
  albumName: propTypes.string.isRequired,
};
