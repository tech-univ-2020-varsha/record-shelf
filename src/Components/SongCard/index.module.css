.songCardContainer{
    display: flex;
    background-color: rgb(210,210,210);
    flex-direction: column;
}

.albumPicture{
    display: flex;
    border-radius:50%;
    padding: 1em;
    max-width: 40%;
    /* align-self: center; */
   
}