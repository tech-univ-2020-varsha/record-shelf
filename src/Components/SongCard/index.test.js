import { render } from '@testing-library/react';
import React from 'react';
import SongCard from '.';

describe('the song list function', () => {
  it('should render card correctly', () => {
    const { asFragment } = render(<SongCard albumArtUrl="url" artists={['a', 'b']} albumName="name" />);
    expect(asFragment).toMatchSnapshot();
  });
});
