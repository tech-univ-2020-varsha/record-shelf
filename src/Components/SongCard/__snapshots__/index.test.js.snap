// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the song list function should render card correctly 1`] = `[Function]`;
